scene/gui/progress_bar.cpp

```diff
@@ -153,14 +213,53 @@ bool ProgressBar::is_percentage_shown() const {
 	return show_percentage;
 }
 
+void ProgressBar::set_indeterminate(bool p_indeterminate) {
+	if (indeterminate == p_indeterminate) {
+		return;
+	}
+	indeterminate = p_indeterminate;
+	_inderminate_fill_progress = 0;
+
+	notify_property_list_changed();
+	set_physics_process_internal(indeterminate);
+	update_minimum_size();
+	queue_redraw();
+}
+
+bool ProgressBar::is_indeterminate() const {
+	return indeterminate;
+}
+
+void ProgressBar::set_editor_preview_indeterminate(bool p_preview_indeterminate) {
+	if (editor_preview_indeterminate == p_preview_indeterminate) {
+		return;
+	}
+	editor_preview_indeterminate = p_preview_indeterminate;
+	_inderminate_fill_progress = 0;
```

Comment:
This will reset the value even at runtime. Maybe you should add a check for editor hint.
Or even
```suggestion
	if (!Engine::get_singleton()->is_editor_hint()) {
		return
	}
```
Switching the flag is ok, but that method should have no other effect at runtime.